icelake/src/io/writer_builder.rs

```diff
@@ -142,7 +143,7 @@ impl<L: FileAppenderLayer<DefaultFileAppender>> WriterBuilder<L> {
     pub async fn build_equality_delta_writer(
         self,
         unique_column_ids: Vec<usize>,
-    ) -> Result<EqualityDeltaWriter<impl FileAppenderFactory>> {
+    ) -> Result<EqualityDeltaWriter<FileAppenderBuilder<L>>> {
```

Comment:
It also may can be replaced as 
`pub async fn build_equality_delta_writer<F: FileAppenderFactory<L::R>>(self,unique_column_ids: Vec<usize>) -> Result<EqualityDeltaWriter<F>`

But the WriterBuilder always use FileAppenderBuilder, so I return FileAppenderBuilder directly.

And I'm concerning do we realy need abtract FileAppenderFactory🤔.

Reply:
Yes, maybe we should remove this `FileAppenderFactory` trait given there is not other implementations.